contracts/src/v0.8/llo-feeds/v0.4.0/test/fee-manager/DestinationFeeManager.general.t.sol

```diff
@@ -0,0 +1,242 @@
+    assertEq(newContractBalance, contractBalance - withdrawAmount);
+    //check the admin balance has increased by the correct amount
+    assertEq(newAdminBalance, adminBalance + withdrawAmount);
+  }
+
+  function test_WithdrawNonAdminAddr() public {
+    //simulate a fee
+    mintLink(address(feeManager), DEFAULT_LINK_MINT_QUANTITY);
+
+    //should revert if not admin
+    vm.expectRevert(ONLY_CALLABLE_BY_OWNER_ERROR);
+
+    //withdraw some balance
+    withdraw(address(link), ADMIN, DEFAULT_LINK_MINT_QUANTITY, USER);
+  }
+
+  function test_eventIsEmittedAfterSurchargeIsSet() public {
+    //native surcharge
+    uint64 nativeSurcharge = FEE_SCALAR / 5;
+
+    //expect an emit
+    vm.expectEmit();
+
+    //emit the event that is expected to be emitted
+    emit NativeSurchargeUpdated(nativeSurcharge);
+
+    //set the surcharge
+    setNativeSurcharge(nativeSurcharge, ADMIN);
+  }
+
+  function test_subscriberDiscountEventIsEmittedOnUpdate() public {
+    //native surcharge
+    uint64 discount = FEE_SCALAR / 3;
+
+    //an event should be emitted
+    vm.expectEmit();
+
+    //emit the event that is expected to be emitted
+    emit SubscriberDiscountUpdated(USER, DEFAULT_FEED_1_V3, address(native), discount);
+
+    //set the surcharge
+    setSubscriberDiscount(USER, DEFAULT_FEED_1_V3, address(native), discount, ADMIN);
+  }
+
+  function test_eventIsEmittedUponWithdraw() public {
+    //simulate a fee
+    mintLink(address(feeManager), DEFAULT_LINK_MINT_QUANTITY);
+
+    //the amount to withdraw
+    uint192 withdrawAmount = 1;
+
+    //expect an emit
+    vm.expectEmit();
+
+    //the event to be emitted
+    emit Withdraw(ADMIN, ADMIN, address(link), withdrawAmount);
+
+    //withdraw some balance
+    withdraw(address(link), ADMIN, withdrawAmount, ADMIN);
+  }
+
+  function test_linkAvailableForPaymentReturnsLinkBalance() public {
+    //simulate a deposit of link for the conversion pool
+    mintLink(address(feeManager), DEFAULT_REPORT_LINK_FEE);
+
+    //check there's a balance
+    assertGt(getLinkBalance(address(feeManager)), 0);
+
+    //check the link available for payment is the link balance
+    assertEq(feeManager.linkAvailableForPayment(), getLinkBalance(address(feeManager)));
+  }
+
+  function test_payLinkDeficit() public {
+    //get the default payload
+    bytes memory payload = getPayload(getV2Report(DEFAULT_FEED_1_V3));
+
+    approveNative(address(feeManager), DEFAULT_REPORT_NATIVE_FEE, USER);
+
+    //not enough funds in the reward pool should trigger an insufficient link event
+    vm.expectEmit();
+
+    IDestinationRewardManager.FeePayment[] memory contractFees = new IDestinationRewardManager.FeePayment[](1);
+    contractFees[0] = IDestinationRewardManager.FeePayment(DEFAULT_CONFIG_DIGEST, uint192(DEFAULT_REPORT_LINK_FEE));
+
+    emit InsufficientLink(contractFees);
+
+    //process the fee
+    processFee(contractFees[0].poolId, payload, USER, address(native), 0);
```

Comment:
@Fletch153  we did not use to pass `poolId` here. 
is there anything worth asserting here in the state after running `processFee` regarding `PoolId` ?